pkg/util/ip_reservation_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"net"
+	"testing"
+)
+
+func TestAllIPBlocksAvailable(t *testing.T) {
+	ips := [8]string{"192.168.92.32/29", "192.168.92.40/29", "192.168.92.48/29", "192.168.92.56/29"}
+
+	cidr := "192.168.92.32/27"
+
+	ip, err := GetUnreservedIPBlock(make(map[string]bool), cidr)
+	if err != nil {
+		t.Errorf(err.Error())
+	}
+	if ip != ips[0] {
+		t.Errorf("Expected IP %s to be released from the free IP pool but got IP %s", ips[0], ip)
+	}
+}
+
+func TestSomeIPBlocksAvailable(t *testing.T) {
```

Comment:
these test functions are prime for Table Driven Tests. They are testing same functions with different data:
https://golang.org/wiki/TableDrivenTests